engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/CallCacheDiffActor.scala

```diff
@@ -0,0 +1,244 @@
+    *   },
+    *   "callB": {
+    *     -- information about call B --
+    *   },
+    *   "hashDifferential": [
+    *     {
+    *       "hash key": {
+    *         "callA": -- hash value for call A, or null --,
+    *         "callB": -- hash value for call B, or null --
+    *       }
+    *     },
+    *     ...
+    *   ]
+    * }
+    */
+  private def buildDiffAndRespond(queryA: MetadataQuery,
+                                  queryB: MetadataQuery,
+                                  responseA: MetadataLookupResponse,
+                                  responseB: MetadataLookupResponse,
+                                  replyTo: ActorRef) = {
+
+    val response = MetadataObject(Map(
+      "callA" -> makeCallInfo(queryA, responseA.eventList),
+      "callB" -> makeCallInfo(queryB, responseB.eventList),
+      "hashDifferential" -> diffHashes(responseA.eventList, responseB.eventList)
+    ))
+
+    replyTo ! RequestComplete((StatusCodes.OK, metadataComponentJsonWriter.write(response).asJsObject))
+
+    context stop self
+    stay()
+  }
+
+  /**
+    * Generates the "info" section of callA or callB
+    */
+  private def makeCallInfo(query: MetadataQuery, eventList: Seq[MetadataEvent]): MetadataComponent = {
+    val callKey = MetadataObject(Map(
+      "workflowId" -> MetadataPrimitive(MetadataValue(query.workflowId.toString)),
+      "callFqn" -> MetadataPrimitive(MetadataValue(query.jobKey.get.callFqn)),
+      "jobIndex" -> MetadataPrimitive(MetadataValue(query.jobKey.get.index.getOrElse(-1)))
+    ))
+
+    val allowResultReuse = attributeToComponent(eventList, { _ == CallCachingKeys.AllowReuseMetadataKey }, { _ => "allowResultReuse" })
+    val executionStatus = attributeToComponent(eventList, { _ == CallMetadataKeys.ExecutionStatus })
+
+    List(callKey, allowResultReuse, executionStatus) combineAll
+  }
+
+  /**
+    * Collects events from the list for which the keys verify the keyFilter predicate
+    * and apply keyModifier to the event's key
+    */
+  private def collectEvents(events: Seq[MetadataEvent],
+                            keyFilter: (String  => Boolean),
+                            keyModifier: (String => String)) = events collect {
+    case event @ MetadataEvent(metadataKey @ MetadataKey(_, _, key), _, _) if keyFilter(key) =>
+      event.copy(key = metadataKey.copy(key = keyModifier(key)))
+  }
+
+  /**
+    * Given a list of events, a keyFilter and a keyModifier, returns the associated MetadataComponent.
+    * Ensures that events are properly aggregated together (CRDTs and latest timestamp rule)
+    */
+  private def attributeToComponent(events: Seq[MetadataEvent], keyFilter: (String  => Boolean), keyModifier: (String => String) = identity[String]) = {
+    MetadataComponent(collectEvents(events, keyFilter, keyModifier))
+  }
+
+  /**
+    * Creates the hash differential between 2 list of events
+    */
+  def diffHashes(eventsA: Seq[MetadataEvent], eventsB: Seq[MetadataEvent]): MetadataComponent = {
```

Comment:
Do you mean separately from https://github.com/broadinstitute/cromwell/pull/2363/files#diff-de6bbcf65493c71e06018e1307bca82eR134 ?